lua/terrortown/entities/items/item_base/shared.lua

```diff
@@ -100,9 +100,32 @@ function ITEM:Bought(ply) end
 function ITEM:Initialize() end
 
 ---
--- useable, but do not modify this!
--- @return boolean whether this @{ITEM} is an equipment
+-- Checks whether this items is a valid equipment item.
+-- @note Useable, but do not modify or overwrite this!
+-- @return boolean Return whether this @{ITEM} is an equipment
 -- @realm shared
 function ITEM:IsEquipment()
     return WEPS.IsEquipment(self)
 end
+
+---
+-- Returns the classname of an item. This is often the name of the Lua file or folder containing the files for the item.
+-- @return string The item's class name
+-- @realm shared
+function ITEM:GetClass()
+    return self.ClassName
```

Comment:
This is very confusing, what is the difference between the class name here and the class name returned by the weps function?

Reply:
It's identical. But entities (like weapons) have a function called `wep:GetClass()`. For some reason items didn't have that function and only a variable named `.ClassName` (which I think also exists on entites). In our code we mostly handle items and weapons as similar things, so imho it should be good to have the same basic functions.

The weps function basically checks if the equipment is a table and then returns `.ClassName` or returns `:GetClass()` it the equipment `:IsValid()`